### PR TITLE
CF: 2471 added trysubmit func

### PR DESCRIPTION
**Pull request by ihor-yamshchykov**
Added ability to trigger submit action from outside of the Winterfell component
Added option to hide submit button

### CODE REVIEW

File: src/button.js [hunk 1]
```diff
@@ -9,8 +9,12 @@ class Button extends React.Component {
...
+    const { className, text, ...props } = this.props;
+
     return (
-      <button href="#"
+      <button
+         {...props}
+         href="#"
          className={this.props.className}
          onClick={this.handleClick.bind(this)}>
         {this.props.text}
```
Comment (macku) [inline src/button.js]: should this be updated to `text` ?

File: src/button.js [hunk 2]
```diff
@@ -9,8 +9,12 @@ class Button extends React.Component {
...
 
   render() {
+    const { className, text, ...props } = this.props;
+
     return (
-      <button href="#"
+      <button
+         {...props}
+         href="#"
          className={this.props.className}
```
Comment (macku) [inline src/button.js]: should this be updated to `className` ?

File: src/index.js [hunk 3]
```diff
@@ -6,6 +6,12 @@ var QuestionPanel = require('./questionPanel');
 
 class Winterfell extends React.Component {
 
+  static trySubmit(form) {
```
Comment (macku) [inline src/index.js]: why `trySubmit` ?

Comment (ihor-yamshchykov) [inline src/index.js]: cause it may fail with validation error

File: src/button.js (hunk 1 above)
Comment (ihor-yamshchykov) [inline src/button.js]: fixed

File: src/questionPanel.js [hunk 4]
```diff
@@ -187,6 +187,8 @@ class QuestionPanel extends React.Component {
       );
     });
 
+    const submitButtonStyle = this.props.isSubmitHidden ? { display: 'none' } : null;
```
Comment (bkozera) [inline src/questionPanel.js]: I think this is not needed - we can just pass `this.props.button.disabled` in schema and this will hide butotn

**Comment by macku** [comment]
Moved to  https://github.com/mudynamics/WinterfellCF/pull/3

**Pull request closed without merging**